src/core/column/sumprod.h

```diff
@@ -27,69 +27,69 @@
-  private:
-    Column col_;
-    Groupby gby_;
-    bool is_grouped_;
-    size_t : 56;
-
-  public:
-    SumProd_ColumnImpl(Column &&col, const Groupby& gby, bool is_grouped)
-      : Virtual_ColumnImpl(gby.size(), col.stype()),
-        col_(std::move(col)),
-        gby_(gby),
-        is_grouped_(is_grouped)
-    {
-      xassert(col_.can_be_read_as<T>());
-    }
-
-
-    bool get_element(size_t i, T* out) const override {
-      T result = !SUM; // 0 for `sum()` and 1 for `prod()`
-      T value;     
-      size_t i0, i1;
-      gby_.get_group(i, &i0, &i1);
-
-      if (is_grouped_){
-        size_t nrows = i1 - i0;
-        bool is_valid = col_.get_element(i, &value);
+template <typename T, bool SUM>
+class SumProd_ColumnImpl : public Virtual_ColumnImpl {
+private:
+  Column col_;
+  Groupby gby_;
+  bool is_grouped_;
+  size_t : 56;
+
+public:
+  SumProd_ColumnImpl(Column &&col, const Groupby& gby, bool is_grouped)
+    : Virtual_ColumnImpl(gby.size(), col.stype()),
+      col_(std::move(col)),
+      gby_(gby),
+      is_grouped_(is_grouped)
+  {
+    xassert(col_.can_be_read_as<T>());
+  }
+
+
+  bool get_element(size_t i, T* out) const override {
+    T result = !SUM; // 0 for `sum()` and 1 for `prod()`
+    T value;
+    size_t i0, i1;
+    gby_.get_group(i, &i0, &i1);
+
+    if (is_grouped_){
+      size_t nrows = i1 - i0;
+      bool is_valid = col_.get_element(i, &value);
+      if (is_valid){
+        result = SUM? static_cast<T>(nrows) * value
+                    : ipow(value, nrows);
+      }
+    } else {
+      for (size_t gi = i0; gi < i1; ++gi) {
+        bool is_valid = col_.get_element(gi, &value);
         if (is_valid){
-          result = SUM? static_cast<T>(nrows) * value
-                      : ipow(value, nrows);
-        }
-      } else {
-        for (size_t gi = i0; gi < i1; ++gi) {
-          bool is_valid = col_.get_element(gi, &value);
-          if (is_valid){
-            result = SUM? result + value
-                        : result * value;
-          }
+          result = SUM? result + value
+                      : result * value;
         }
       }
-
-      *out = result;
-      return true; // the result is never a missing value
     }
 
+    *out = result;
+    return true; // the result is never a missing value
+  }
 
-    ColumnImpl *clone() const override {
-      return new SumProd_ColumnImpl(Column(col_), Groupby(gby_), is_grouped_);
-    }
 
+  ColumnImpl *clone() const override {
+    return new SumProd_ColumnImpl(Column(col_), Groupby(gby_), is_grouped_);
+  }
 
-    size_t n_children() const noexcept override {
-      return 1;
-    }
 
+  size_t n_children() const noexcept override {
+    return 1;
+  }
 
-    const Column &child(size_t i) const override {
-      xassert(i == 0);
-      (void)i;
-      return col_;
-    }
-  };
+
+  const Column &child(size_t i) const override {
+    xassert(i == 0);
+    (void)i;
+    return col_;
+  }
+};
```

Comment:
The changes above are not relevant to your issues, I have just fixed indentation in that file.